cmake/modules/AddCIRCT.cmake

```diff
@@ -0,0 +1,43 @@
+include_guard()
+
+function(add_circt_dialect dialect dialect_namespace)
+  add_mlir_dialect(${ARGV})
+  add_dependencies(circt-headers MLIR${dialect}IncGen)
+endfunction()
```

Comment:
It gives me the heebie-jeebies to see this stuff, which is going to get replicated in many many downstream MLIR projects.  I'm not opposed to adding this temporarily, but I'd really rather see it get factored out so that any MLIR-dependent project can instantiate this boilerplate.